### PR TITLE
Twitter a

### PR DESCRIPTION
**Pull request by moses-alexander**

### CODE REVIEW

**Comment by moses-alexander** [comment]
WIP PR ... this module is 99% complete, but for some reason django templates won't display model attributes or load them in page source despite verification of POST request and model instantiation

File: apps/twitter_a/templates/twitter_a/results.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	<thead>
+	  <tr>
+		<th>Tweet ID</th>
+		<th>Tweet Text</th>
+		<th>Tweet Date</th>
+		<th>Tweet URL</th>
+	  </tr>
+	</thead>
+	<tbody>
+        { % for val in vals % }
```
Comment (karimbahgat) [inline apps/twitter_a/templates/twitter_a/results.html:27]: The problem might be the space between curly bracket and percent sign, it should be without space, ie {% for val in vals %}.

File: apps/twitter_a/templates/twitter_a/results.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        <tr>
+            <td> {{ val.source_id }} </td>
+            <td> {{ val.source_text }} </td>
+            <td> {{ val.source_date }} </td>
+            <td> {{ val.source_url }} </td>
+        </tr>
+        { % endfor % }
```
Comment (karimbahgat) [inline apps/twitter_a/templates/twitter_a/results.html:34]: Same here

File: apps/twitter_a/templates/twitter_a/results.html (hunk 1 above)
Comment (moses-alexander) [reply apps/twitter_a/templates/twitter_a/results.html]: thanks, this solved it!

**Comment by moses-alexander** [comment]
@karimbahgat @DanRunfola final commits added, module is ready to be tested and merged.

**Comment by moses-alexander** [comment]
one note, the API creds in api.txt might be invalid. the keys are labelled so it should be straightforward to replace them with the academic dev account keys that Dan has. also note that the "Full Archive Sandbox Environment" needs to be enabled on that account and the environment label needs to be edited in the code (https://github.com/wmgeolab/scope/blob/7035726aa295c7d307384c53f891fc49d766fb1f/apps/twitter_a/twitter_search.py#L137).

image

image

image

The results.html page takes ~30-45 sec to load when retrieving 100 tweets (the max allowed by twitter api).

**Pull request closed without merging**